### ISSUE TITLE
Popover - Ensure the focus order of interactive elements on the page is logical - (2036570079)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-design-system#7808]
### Violation: 
 Ensure the focus order of interactive elements on the page is logical 

![image](https://github.com/Esri/calcite-design-system/assets/5023024/8ac9e988-7892-454f-8f55-5c477c2d0ced)

### WCAG Reference: 
- [2.4.3 Focus Order](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html) 

### Severity: 
 6 

### Media Type: 
 Keyboard Accessibility 

--- 
### Areas for Remediation: 
- Title: Popover - Ensure the focus order of interactive elements on the page is logical - (2036570079) 
- Module: Popover 
**Issue**
When user activates the "Activate Popover with Close Button" focus order is illogical:
1. The X(close) button that is available when activates the "Popover with Close Button" that operable after shift tab. 
2. Activate Popover with Close Button 

**User Impact**
Keyboard-only users will have difficulty efficiently navigating the site.

**Code Reference** 
 ```
<button aria-busy="false" aria-disabled="false" aria-label="Close" class="button"><div aria-hidden="true" class="icon-container"><div class="slot-container"><slot></slot></div></div><div class="text-container">Close</div></button>
(...)
<calcite-button id="popover-button2" alignment="center" appearance="solid" color="blue" scale="m" width="auto" calcite-hydrated="" aria-controls="calcite-popover-c3a8eae9-e0eb-1875-1b3e-06a9d4c47a2e" aria-expanded="true">Activate Popover with Close Button</calcite-button>
 ``` 
**Suggestion**
Ensure the focus order of interactive elements on the page is logical. The best way to set the focus order is to reorder the elements in the DOM.
Correct focus order:
1. Activate Popover with Close Button
2. The X(close) button that is available when activates the "Popover with Close Button".

**Compliant Code Example** 
 ```
<calcite-button id="popover-button2" alignment="center" appearance="solid" color="blue" scale="m" width="auto" calcite-hydrated="" aria-controls="calcite-popover-c3a8eae9-e0eb-1875-1b3e-06a9d4c47a2e" aria-expanded="true">Activate Popover with Close Button</calcite-button>
(...)
<button aria-busy="false" aria-disabled="false" aria-label="Close" class="button"><div aria-hidden="true" class="icon-container"><div class="slot-container"><slot></slot></div></div><div class="text-container">Close</div></button>
 ```
 
--- 
### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by driskull** [comment]
I think this is just an app problem.

**Comment by geospatialem** [comment]
> I think this is just an app problem.

Agree. This could be doc'd for folks to know of under the component's [accessibility](https://developers.arcgis.com/calcite-design-system/components/popover/#accessibility) section. Going to modify this issue to be a doc update. 👍🏻

**Comment by geospatialem** [comment]
For verification, language has been added to the [Popover](https://developers.arcgis.com/calcite-design-system/components/popover/) component page as part of the 1/12/24 release.

**Comment by geospatialem** [comment]
Verified on production in the [Popover's accessibility](https://developers.arcgis.com/calcite-design-system/components/popover/#accessibility) page section.

![image](https://github.com/Esri/calcite-design-system/assets/5023024/cf2f7470-5d22-4615-a475-784114ada897)
